### DISCUSSION TITLE
gomod PRs for individual packages?

### DISCUSSION THREAD

**Discussion by caphrim007** [renovatebot/renovate#8619] [Request Help]
I'm trying to replicate a dependency updating experience similar to what is seen in this test repo

https://github.com/rarkins/lazygit-example

I had enabled such a config, but when I run it against my bitbucket repo, it creates many PRs, but they all have the same changes; there are not individual go.mod and go.sum changes in each PR.

Each PR though does have a unique name.

Is there any chance this has something to do with this change?

https://github.com/renovatebot/renovate/blob/master/lib/manager/gomod/artifacts.ts#L77

When I run that command locally, I observe the behavior that I see in the PR. All the changes instead of just 1 package change.

**Comment by viceice** [comment]
Please provide your full repo and bot admin config (credentials sanmitized of cause) 

Also a [minimal reproduction](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) repo helps too

**Reply by caphrim007** [reply]
full repo is
```
{
  "extends": [
    "config:base"
  ],
  "prHourlyLimit": 100
}
```


bot admin is

```
    module.exports = {
      autodiscover: false,
      platform: 'bitbucket',
      gitAuthor: 'Renovate Bot,
      onboarding: false,
      dryRun: false,
      repositories: ["myorg/repo"],
      customEnvVariables: {
        "GOCACHE": "/scratch/.cache"
      }
    };
```

In terms of repo I copied what is at rarkins example and just put the same files in a bitbucket org. I added a bitbucket.org private repo as I was in the process of making SSH things work, but that is, otherwise, the only change I made to the go.mod file.

```
require (
        bitbucket.org/myorg/myrepo v0.0.0-20201106173536-93b7ae6971b8
        ....rest is same as rarkins file
)

**Reply by caphrim007** [reply]
did some more digging and it seems that bitbucket.org repo is entirely responsible. when i remove it, things appear like they normally do

trying to dig into why that's the case.

**Reply by viceice** [reply]
Set `LOG_LEVEL=debug` to the bot env, so you see more relevant information what happen.

**Reply by caphrim007** [reply]
I created a bitbucket repo with only a go.mod file in it and added the following to that file

```
module bitbucket.org/caphrim007-bitbucket/test

go 1.13

require (
	github.com/antlr/antlr4 v0.0.0-20201129091037-eafcc9369e99
	github.com/aws/aws-sdk-go v1.19.8
	github.com/caio/go-tdigest v2.3.0+incompatible // indirect
	github.com/go-kit/kit v0.8.0
	github.com/go-redis/redis/v8 v8.3.4
	github.com/golang/protobuf v1.4.2
	github.com/google/uuid v1.1.1
	github.com/klauspost/compress v1.9.8 // indirect
	github.com/leesper/go_rng v0.0.0-20190531154944-a612b043e353 // indirect
	github.com/pkg/errors v0.9.1
	github.com/prometheus/client_golang v1.1.0
	github.com/prometheus/client_model v0.2.0
	github.com/satori/go.uuid v1.2.0
	github.com/stretchr/testify v1.6.1
	github.com/wavefronthq/wavefront-sdk-go v0.9.3
	github.com/xdg/stringprep v1.0.1-0.20180714160509-73f8eece6fdc // indirect
	go.mongodb.org/mongo-driver v1.3.0
	golang.org/x/exp v0.0.0-20200821190819-94841d0725da // indirect
	golang.org/x/net v0.0.0-20201006153459-a7d1128ccaa0
	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45
	gonum.org/v1/gonum v0.7.0 // indirect
	google.golang.org/appengine v1.6.5 // indirect
	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
)
```

With the exception of the module definition at the top, this is the same thing that is in the private BB repo that seems to be giving me my issue.

Since there's still some ongoing work in renovate to fully describe BB stuff with keys and all that jazz, maybe you can put that in a github repo to see if you have a similar experience.

With that single file in place, I adjusted my BB require line on my main repo under renovate control, and I get the behavior I originally described; many prs all with the same content.

**Reply by caphrim007** [reply]
for reference, the lines I now have in my renovated repo now look like this

```
require (
	bitbucket.org/caphrim007-bitbucket/test 5f0800e
	github.com/aws/aws-sdk-go v1.15.21
         ...more non-bitbucket stuff here
```

The caphrim007-bitbucket/test repo also has a go.mod which i pasted in the previous comment

**Reply by caphrim007** [reply]
Also, I have the debug env variable set (i normally run it with debug output on) if there's something in particular that you want me to look for or paste? nothing appears out of place from what I can see, but lemme know.

**Reply by viceice** [reply]
Sorry, we don't have time to create a minimal reproduction ourself, so please create a public one, can be on bitbucket.org.

**Reply by caphrim007** [reply]
A bitbucket reproduction is here

https://bitbucket.org/caphrim007-bitbucket/golang-test1/
https://bitbucket.org/caphrim007-bitbucket/golang-test2/

Both are public repos.

I've also created public repos on github

https://github.com/caphrim007/golang-test1
https://github.com/caphrim007/golang-test2

As an example, I'll use this PR that renovate generated

https://github.com/caphrim007/golang-test1/pull/6

The package is `github.com/kevinburke/ssh_config`. Looking at the content of this package, seen here, https://github.com/kevinburke/ssh_config there is no go.mod or further set of dependencies. Therefore, I would expect the renovate PR to include an update for ssh_config only, and a change to the go.sum for the ssh_config updates only.

Instead, the PR that is generated includes updates for 15 some packages, and go.sum updates for....too many to list

I can't make heads-or-tails as to why that is. And it appears to be repeated across all of the PRs.

app.renovate run for this appears to be here

https://app.renovatebot.com/dashboard#github/caphrim007/golang-test1/298877746

The affect only seems to happen in cases where a go.sum file exists. If there is no go.sum, then renovate emits a message saying that no go.sum file was found, and it only updates the go.mod file (correctly I might add).

Hope that provides more insight.

**Reply by caphrim007** [reply]
another run is here in case you can view it via some internal means.

https://app.renovatebot.com/dashboard#github/caphrim007/golang-test1/298877830

**Comment by rarkins** [comment]
Where/how did you find that repo you copied off? I don't recall when/why it was made.

**Reply by caphrim007** [reply]
@rarkins https://www.whitesourcesoftware.com/free-developer-tools/blog/automating-go-module-dependency-updates/

url is at the bottom

**Reply by rarkins** [reply]
Thanks, looks like it needs updating!

**Reply by caphrim007** [reply]
i dont know that the article is wrong per-se. I forked the repo here and you can see that the PRs that it opened all look correct (afaik)

https://github.com/caphrim007/lazygit-example/pulls

I think there's just something awkward about the repo containing the one additional requirement. I'm going to try to reproduce with 2 github repos. @viceice the repo here is public

bitbucket.org/caphrim007-bitbucket/test

However, as I understand BB support in renovate, it's not officially signed off as fully baked, so while I will attempt to make a reliable repro with it, I may not be able to.
